types/wlr_output.c

```diff
@@ -377,25 +374,16 @@ void wlr_output_make_current(struct wlr_output *output) {
 
 void wlr_output_swap_buffers(struct wlr_output *output) {
 	if (output->cursor.is_sw) {
-		glViewport(0, 0, output->width, output->height);
-		glEnable(GL_BLEND);
-		glBlendFunc(GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
-
-		struct wlr_texture *texture = output->cursor.texture;
-		struct wlr_renderer *renderer = output->cursor.renderer;
-		if (output->cursor.surface) {
-			texture = output->cursor.surface->texture;
-			renderer = output->cursor.surface->renderer;
-		}
+		struct wlr_tex *tex = output->cursor.tex;
```

Comment:
This texture can be NULL or invalid, if the cursor surface has been destroyed or an error has occurred.